caddytls/setup.go

```diff
@@ -66,6 +66,12 @@ func setupTLS(c *caddy.Controller) error {
 		for c.NextBlock() {
 			hadBlock = true
 			switch c.Val() {
+			case "ca":
+				arg := c.RemainingArgs()
```

Comment:
I've replaced the check with `len(args) != 1` similar to how `dns` and `storage` check their args, and added test cases for too many/too few args.